avocado/core/runner.py

```diff
@@ -194,32 +194,38 @@ def _add_status_failures(self, test_state):
                                       " see overall job.log for details.")
         return test_state
 
-    def finish(self, proc, started, timeout, step):
+    def finish(self, proc, started, step, deadline, result_dispatcher):
         """
         Wait for the test process to finish and report status or error status
         if unable to obtain the status till deadline.
 
         :param proc: The test's process
         :param started: Time when the test started
-        :param timeout: Timeout for waiting on status
         :param first: Delay before first check
         :param step: Step between checks for the status
+        :param deadline: The available deadline
```

Comment:
I don't get the idea of *available* deadline.  Maybe you meant the deadline "set", as in, "The deadline set for test process status reporting".